### PR TITLE
allows hashtag hrefs in sanitize

### PR DESCRIPTION
**Pull request by tomlum**
Allows data hrefs that start with a hashtag (#) to pass through SVG sanitizing

### CODE REVIEW

File: src/sanitize-svg.js [hunk 1]
```diff
@@ -11,19 +11,22 @@ DOMPurify.addHook(
     'beforeSanitizeAttributes',
     currentNode => {
 
-        if (currentNode && currentNode.href && currentNode.href.baseVal &&
-            currentNode.href.baseVal.replace(/\s/g, '').slice(0, 5) !== 'data:'){
+        if (currentNode && currentNode.href && currentNode.href.baseVal){
```
Comment (benjiwheeler) [inline src/sanitize-svg.js]: This is super tiny, but: our general code style is to have a space between the closing parenthesis and the opening curly brace
```suggestion
        if (currentNode && currentNode.href && currentNode.href.baseVal) {
```

File: src/sanitize-svg.js [hunk 2]
```diff
@@ -11,19 +11,22 @@ DOMPurify.addHook(
...
+            const href = currentNode.href.baseVal.replace(/\s/g, '');
+            // "data:" and "#" are valid hrefs
+            if ((href.slice(0, 5) !== 'data:') && (href.slice(0, 1) !== '#')){
```
Comment (benjiwheeler) [inline src/sanitize-svg.js]:
```suggestion
            if ((href.slice(0, 5) !== 'data:') && (href.slice(0, 1) !== '#')) {
```

File: src/sanitize-svg.js [hunk 3]
```diff
@@ -11,19 +11,22 @@ DOMPurify.addHook(
...
 
-            if (currentNode.attributes.getNamedItem('xlink:href')){
-                currentNode.attributes.removeNamedItem('xlink:href');
-                delete currentNode['xlink:href'];
-            }
-            if (currentNode.attributes.getNamedItem('href')){
-                currentNode.attributes.removeNamedItem('href');
-                delete currentNode.href;
+                if (currentNode.attributes.getNamedItem('xlink:href')){
```
Comment (benjiwheeler) [inline src/sanitize-svg.js]:
```suggestion
                if (currentNode.attributes.getNamedItem('xlink:href')) {
```

File: src/sanitize-svg.js [hunk 4]
```diff
@@ -11,19 +11,22 @@ DOMPurify.addHook(
...
+                    currentNode.attributes.removeNamedItem('xlink:href');
+                    delete currentNode['xlink:href'];
+                }
+                if (currentNode.attributes.getNamedItem('href')){
```
Comment (benjiwheeler) [inline src/sanitize-svg.js]:
```suggestion
                if (currentNode.attributes.getNamedItem('href')) {
```

File: src/sanitize-svg.js [hunk 5]
```diff
@@ -11,19 +11,22 @@ DOMPurify.addHook(
...
+                    currentNode.attributes.removeNamedItem('href');
+                    delete currentNode.href;
+                }
             }
+            return currentNode;
```
Comment (benjiwheeler) [inline src/sanitize-svg.js]: Why was this `return` moved one level in? Shouldn't we always be returning it?

**Review by benjiwheeler — CHANGES_REQUESTED** [review]

File: src/sanitize-svg.js (hunk 5 above)
Comment (tomlum) [reply src/sanitize-svg.js]: ah yes, good catch!

**Comment by benjiwheeler** [comment]
LGTM! I added a test for this, by putting before/after file versions in the fixtures dir

**Pull request merged**